File: backend/member-service/src/main/java/com/example/memberservice/profile/dto/ProfileDto.java
package com.example.memberservice.profile.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProfileDto implements Serializable {

    private Long id;
    private Long memberId;
    private ProfileType profileType;
    private String value;
    private String createdDate;
    private String updatedDate;

    public static ProfileDto from(Long id, Long memberId, ProfileType profileType, String value, String createdDate, String updatedDate) {
        return ProfileDto.builder()
                .id(id)
                .memberId(memberId)
                .profileType(profileType)
                .value(value)
                .createdDate(createdDate)
                .updatedDate(updatedDate)
                .build();
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/profile/dto/ProfileDto.kt
package com.example.memberservice.profile.dto

import java.io.Serializable

data class ProfileDto(
    var id: Long? = null,
    var memberId: Long? = null,
    var profileType: ProfileType? = null,
    var value: String? = null,
    var createdDate: String? = null,
    var updatedDate: String? = null,
) : Serializable {
    companion object {
        @JvmStatic
        fun from(id: Long?, memberId: Long?, profileType: ProfileType?, value: String?, createdDate: String?, updatedDate: String?): ProfileDto =
            ProfileDto(id, memberId, profileType, value, createdDate, updatedDate)
    }
}
